### PR TITLE
SlimefunItem : renames of some methods, made ghost configurable, changed "name" to "id" and privatized all fields

### PR DESCRIPTION
**Pull request by Poslovitch**
This PR adds or modifies to the SlimefunItem and its derivative classes : 
- all fields are now private, and can only be accessed using a method
- "name" is now "ID"
- some documentation has been added (really succinct, eg. only the version a method has been added, etc)
- "ghost" is now "hidden", as well as being configurable in Items.yml with "hide-in-guide"
- some fields are initialized inline instead of constructors (to make reading a bit easier)

### CODE REVIEW

File: src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java [hunk 1]
```diff
@@ -508,20 +509,12 @@ public void registerChargeableBlock(boolean vanilla, int capacity, ItemHandler..
...
 		return BlockStorage.getInventory(b);
 	}
 	
 	public void addWikipage(String page) {
-		Slimefun.addWikiPage(this.getName(), "https://github.com/mrCookieSlime/Slimefun4/wiki/" + page);
-	}
-	
-	public boolean isAddonItem() {
-		return this.addon;
+		Slimefun.addWikiPage(this.getID(), "https://github.com/mrCookieSlime/Slimefun4/wiki/" + page);
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java]: Not related to this PR, but it's TheBusyBiscuit now.

File: src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java [hunk 2]
```diff
@@ -208,52 +224,55 @@ public void register(boolean slimefun) {
...
 					handlers.put(handler.toCodename(), handlerset);
 				}
 				
-				if (SlimefunStartup.getCfg().getBoolean("options.print-out-loading")) System.out.println("[Slimefun] Loaded Item \"" + this.getName() + "\"");
+				if (SlimefunStartup.getCfg().getBoolean("options.print-out-loading")) System.out.println("[Slimefun] Loaded Item \"" + this.getID() + "\"");
 			} else {
-			    if (this instanceof VanillaItem) this.state = State.VANILLA;
-			    else this.state = State.DISABLED;
+			    if (this instanceof VanillaItem) state = State.VANILLA;
+			    else state = State.DISABLED;
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java]: Do not remove all the 'this.' statements, they are not needed but make it cleaner, clarifying it is a class level variable.

File: src/me/mrCookieSlime/Slimefun/api/Slimefun.java [hunk 3]
```diff
@@ -184,16 +184,16 @@ public static boolean hasUnlocked(Player p, SlimefunItem sfItem, boolean message
 	 */
 	public static boolean hasPermission(Player p, SlimefunItem item, boolean message) {
 		if (item == null) return true;
-		else if (SlimefunStartup.getItemCfg().getString(item.getName() + ".required-permission").equalsIgnoreCase("")) return true;
-		else if (p.hasPermission(SlimefunStartup.getItemCfg().getString(item.getName() + ".required-permission"))) return true;
+		else if (item.getPermission().equalsIgnoreCase("")) return true;
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/api/Slimefun.java]: A nice performance improvement.

Comment (Poslovitch) [reply src/me/mrCookieSlime/Slimefun/api/Slimefun.java]: Thanks :)

File: src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java (hunk 1 above)
Comment (Poslovitch) [reply src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java]: Then, it's time to fix it :D

File: src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java (hunk 2 above)
Comment (Poslovitch) [reply src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java]: Gotcha :)

**Comment by Poslovitch** [comment]
All the "renamed" methods are new, and the old one are deprecated.

**Comment by Poslovitch** [comment]
@TheBusyBiscuit when do you plan to merge it : I have got some other PR waiting this to be merged to avoid conflicts

**Pull request merged**